lib/libc/Kconfig

```diff
@@ -6,19 +6,41 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+
 menu "C Library"
-depends on !NATIVE_APPLICATION
+
+choice LIB_C_IMPLEMENTATION
+	prompt "C Library Implementation"
+	default MINIMAL_LIBC
+
+config MINIMAL_LIBC
+	bool "Minimal c library"
+	depends on !NATIVE_APPLICATION
+	depends on !SOC_SERIES_STM32H7X
```

Comment:
I would prefer to see the STM32H7X changes pulled out to a separate commit.

Reply:
done